src/cls/rgw/cls_rgw_ops.h

```diff
@@ -1002,56 +1002,86 @@ struct cls_rgw_lc_get_next_entry_op {
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_op)
 
 struct cls_rgw_lc_get_next_entry_ret {
-  pair<string, int> entry;
+private:
+  mutable pair<string, int> compat_entry; // only for backward compatibility
+public:
+  cls_rgw_lc_entry entry;
 
   cls_rgw_lc_get_next_entry_ret() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
+    compat_entry = make_pair(entry.bucket, entry.status);
+    encode(compat_entry, bl); 
     encode(entry, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(bufferlist::iterator& bl) {
-    DECODE_START(1, bl);
-    decode(entry, bl);
+    DECODE_START(2, bl);
+    decode(compat_entry, bl);
+    entry.bucket = compat_entry.first;
+    entry.status = compat_entry.second;
+    if (struct_v >= 2) {
+      decode(entry, bl);
+    }
     DECODE_FINISH(bl);
   }
 
 };
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_ret)
 
 struct cls_rgw_lc_rm_entry_op {
-  pair<string, int> entry;
+private:
+  mutable pair<string, int> compat_entry; // only for backward compatibility
+public:
+  cls_rgw_lc_entry entry;
   cls_rgw_lc_rm_entry_op() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
+    compat_entry = make_pair(entry.bucket, entry.status);
+    encode(compat_entry, bl); 
     encode(entry, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(bufferlist::iterator& bl) {
-    DECODE_START(1, bl);
-    decode(entry, bl);
+    DECODE_START(2, bl);
+    decode(compat_entry, bl);
+    entry.bucket = compat_entry.first;
+    entry.status = compat_entry.second;
+    if (struct_v >= 2) {
+      decode(entry, bl);
+    }
     DECODE_FINISH(bl);
   }
 };
 WRITE_CLASS_ENCODER(cls_rgw_lc_rm_entry_op)
 
 struct cls_rgw_lc_set_entry_op {
-  pair<string, int> entry;
+private:
+  mutable pair<string, int> compat_entry; // only for backward compatibility
```

Comment:
no need to have compat_entry member, use local variables when you encode and decode